pkg/apis/workflow/v1alpha1/object_types.go

```diff
@@ -0,0 +1,24 @@
+package v1alpha1
+
+import (
+	"encoding/json"
+)
+
+// +kubebuilder:validation:Type=object
+type Object struct {
```

Comment:
I don't think `Object` is the right name for this. At the very least it should signify that it is used as a template type (e.g. `AbstractTemplate`), or better that it is used as a stand-in for a plugin (e.g. `Plugin`).

Reply:
I get it, but the type is a generic object, maybe...?

```golang
type PluginTemplate = Object
````

Reply:
Yes that is the correct solution imo